File: receptor/capas/enlace.js
/** Capa de enlace: calcula y verifica la integridad delegando en el algoritmo del registro. */

import { obtenerAlgoritmo } from "../algoritmos/registro.js";

/** @param {string} bits @param {string} nombreAlgoritmo @returns {import("../algoritmos/contrato.js").TramaCodificada} */
export function calcularIntegridad(bits, nombreAlgoritmo, configuracion = {}) {
  return obtenerAlgoritmo(nombreAlgoritmo, configuracion).calcular(bits);
}

/** @param {string} trama @param {string} nombreAlgoritmo @param {Object} parametros @returns {import("../algoritmos/contrato.js").ResultadoVerificacion} */
export function verificarIntegridad(trama, nombreAlgoritmo, parametros) {
  return obtenerAlgoritmo(nombreAlgoritmo).verificar(trama, parametros);
}

File: emisor/capas/enlace.py
"""Capa de enlace: calcula y verifica la integridad delegando en el algoritmo del registro."""

from algoritmos.contrato import ResultadoVerificacion, TramaCodificada
from algoritmos.registro import obtener_algoritmo


def calcular_integridad(bits: str, nombre_algoritmo: str) -> TramaCodificada:
    """Delega en el algoritmo del registro y devuelve la trama con redundancia."""
    return obtener_algoritmo(nombre_algoritmo).calcular(bits)


def verificar_integridad(trama: str, nombre_algoritmo: str, parametros: dict) -> ResultadoVerificacion:
    """Verifica (y corrige si el algoritmo puede) la trama recibida."""
    return obtener_algoritmo(nombre_algoritmo).verificar(trama, parametros)
